backend/app/assets/stylesheets/spree/backend/components/_navigation.scss

```diff
@@ -69,40 +133,54 @@ nav.menu {
 
     &.selected {
       background: $color-navbar-active-bg;
+      @include media($desktop) {
+        position: relative;
+      }
 
-      > a {
-        color: $color-navbar-active;
+      &:not(:hover) > ul {
+        @include media($desktop) {
+          display: none;
+        }
+        @include media($tablet) {
+          display: block;
+        }
       }
-    }
 
-    &.selected li {
-      padding-left: 2em;
-    }
+      > ul {
+        @include media($desktop) {
+          @include flyout;
```

Comment:
A small screen does not equal a touch screen and vice versa. And this PR actually added more code to make this work "only" for desktop. That's why I proposed to remove this refactoring and leave the flyout even for small screens.